### ISSUE TITLE
building on SmartOS

### ISSUE THREAD

**Issue by liftedkilt** [CumulusNetworks/quagga#18]
[Line 25](https://github.com/CumulusNetworks/quagga/blob/cmaster/README.NetBSD#L25) of the netbsd/pkgsrc build instructions is failing on Solaris 11

[root: ~/quagga]# uname -a
SunOS 7a89e59e-cbbd-444b-936b-6c0655c954c4 5.11 joyent_20160915T211515Z i86pc i386 i86pc Solaris

[root: ~/quagga]# ./bootstrap.sh 
Makefile.am:29: warning: variable 'ACLOCAL_LDFLAGS' is defined but no program or
Makefile.am:29: library has 'ACLOCAL' as canonical name (possible typo)
lib/Makefile.am:7: error: Libtool library used but 'LIBTOOL' is undefined
lib/Makefile.am:7:   The usual way to define 'LIBTOOL' is to add 'LT_INIT'
lib/Makefile.am:7:   to 'configure.ac' and run 'aclocal' and 'autoconf' again.
lib/Makefile.am:7:   If 'LT_INIT' is in 'configure.ac', make sure
lib/Makefile.am:7:   its definition is in aclocal's search path.
ospfclient/Makefile.am:6: error: Libtool library used but 'LIBTOOL' is undefined
ospfclient/Makefile.am:6:   The usual way to define 'LIBTOOL' is to add 'LT_INIT'
ospfclient/Makefile.am:6:   to 'configure.ac' and run 'aclocal' and 'autoconf' again.
ospfclient/Makefile.am:6:   If 'LT_INIT' is in 'configure.ac', make sure
ospfclient/Makefile.am:6:   its definition is in aclocal's search path.
ospfd/Makefile.am:8: error: Libtool library used but 'LIBTOOL' is undefined
ospfd/Makefile.am:8:   The usual way to define 'LIBTOOL' is to add 'LT_INIT'
ospfd/Makefile.am:8:   to 'configure.ac' and run 'aclocal' and 'autoconf' again.
ospfd/Makefile.am:8:   If 'LT_INIT' is in 'configure.ac', make sure
ospfd/Makefile.am:8:   its definition is in aclocal's search path.
solaris/Makefile.am:71: warning: '%'-style pattern rules are a GNU make extension
solaris/Makefile.am:75: warning: '%'-style pattern rules are a GNU make extension
solaris/Makefile.am:79: warning: '%'-style pattern rules are a GNU make extension
solaris/Makefile.am:84: warning: '%'-style pattern rules are a GNU make extension
solaris/Makefile.am:88: warning: '%'-style pattern rules are a GNU make extension
solaris/Makefile.am:98: warning: '%'-style pattern rules are a GNU make extension
solaris/Makefile.am:103: warning: '%'-style pattern rules are a GNU make extension
autoreconf: automake failed with exit status: 1

**Comment by donaldsharp** [comment]
A quick google search:

http://stackoverflow.com/questions/22526375/where-is-aclocals-search-path

TL;DR - it is looking for the aclocal program in your path.  Do you have it installed correctly?

**Comment by mwinter-osr** [comment]
It looks to me like pkg-config is missing.

Here is the packages I install for my building on OmniOS (which is a solaris distro):

	pkg install \
	  developer/build/autoconf \
	  developer/build/automake \
	  developer/lexer/flex \
	  developer/parser/bison \
	  developer/object-file \
	  developer/linker \
	  developer/library/lint \
	  developer/build/gnu-make \
	  developer/gcc51 \
	  library/idnkit \
	  library/idnkit/header-idnkit \
	  system/header \
	  system/library/math/header-math \
	  git libtool gawk pkg-config	

	pkgadd -d http://get.opencsw.org/now
	/opt/csw/bin/pkgutil -U
	/opt/csw/bin/pkgutil -y -i texinfo
	/opt/csw/bin/pkgutil -y -i perl
	/opt/csw/bin/pkgutil -y -i libjson_c_dev

**Comment by liftedkilt** [comment]
Thank you both for your input. I kept poking at it and was unable to get it working, but at this point it looks like an avenue we aren't going to keep putting resources towards. The SmartOS folks are steering us away from attempting to use quagga for the underlay network in the Triton SmartDataCenter product.

I may poke at this a bit more, but since I'm not going to be pursuing the concept actively, I'll just close this issue.

Thanks
